service/create/master_service.go

```diff
@@ -0,0 +1,116 @@
+package create
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awsutil "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/awstpr"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/microerror"
+)
+
+type MasterServiceInput struct {
+	Clients   awsutil.Clients
+	Cluster   awstpr.CustomObject
+	MasterIDs []string
+}
+
+func (s *Service) createMasterService(input MasterServiceInput) error {
+	for _, masterID := range input.MasterIDs {
+		reservations, err := input.Clients.EC2.DescribeInstances(&ec2.DescribeInstancesInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String("instance-id"),
+					Values: []*string{
+						aws.String(masterID),
+					},
+				},
+			},
+		})
```

Comment:
We're getting rid of the old AWS resources but it looks like it supports finding instances. Could you use that?

https://github.com/giantswarm/aws-operator/blob/e796869c17eb3a40322c3197463adf529b8d5cdd/resources/aws/instance.go#L237

Reply:
Yeah sweet, I didn't look if we had a good way for this already. Will update.